### PR TITLE
fix:banner ~ adds mdn plus launch banner

### PR DESCRIPTION
**Pull request by schalkneethling**
Adds MDN Plus launch banner

Screenshot of the MDN Plus launch banner. The banner contains the words, "MDN Plus is here! Support MDN and make it your own. Learn more"

### CODE REVIEW

File: client/src/banners/active-banner.tsx [hunk 1]
```diff
@@ -71,24 +71,25 @@ function SendCTAEventToGA(eventCategory: string) {
...
 
-function RedesignAnnouncementBanner({
+function PlusLaunchAnnouncementBanner({
   onDismissed,
 }: {
   onDismissed: () => void;
 }) {
   return (
-    <Banner id={REDESIGN_ANNOUNCEMENT} onDismissed={onDismissed}>
+    <Banner id={PLUS_LAUNCH_ANNOUNCEMENT} onDismissed={onDismissed}>
```
Comment (caugner) [inline client/src/banners/active-banner.tsx:80]: Shouldn't this banner only show if `PLUS_IS_ENABLED` **and** available in the user's region (or do we only know the user's region once they're logged in)?

```tsx
    {PLUS_IS_ENABLED && (<Banner id={PLUS_LAUNCH_ANNOUNCEMENT} onDismissed={onDismissed}>
```

**Review by caugner — CHANGES_REQUESTED** [review]
If we add the banner behind the `PLUS_IS_ENABLED` flag, we just have to flip that flag later.

File: client/src/banners/active-banner.tsx [hunk 2]
```diff
@@ -105,8 +108,10 @@ export default function ActiveBanner({
   id: string;
   onDismissed: () => void;
 }) {
-  if (id === REDESIGN_ANNOUNCEMENT) {
-    return <RedesignAnnouncementBanner onDismissed={onDismissed} />;
+  const userData = useUserData();
+
+  if (id === PLUS_LAUNCH_ANNOUNCEMENT && isPlusAvailable(userData)) {
```
Comment (schalkneethling) [inline client/src/banners/active-banner.tsx:113]: @caugner Ok, I added the plus available check here.

File: client/src/banners/active-banner.tsx (hunk 1 above)
Comment (schalkneethling) [reply client/src/banners/active-banner.tsx:82]: > or do we only know the user's region once they're logged in

Oh dang, that is a good question. I have a feeling we probably only know that after they are logged in. @Guyzeroth do you know?

Comment (schalkneethling) [reply client/src/banners/active-banner.tsx:82]: I am actually 100% sure about that. There is no way we are storing that in any other way.

Comment (schalkneethling) [reply client/src/banners/active-banner.tsx:82]: Unless I am wrong, the only other option is to base it on the `locale`. That does mean a user can simply change their locale, see the banner, etc. They will still not be able to sign up unless they have an appropriate card.

**Review by Guyzeroth — APPROVED** [review]

File: client/src/banners/active-banner.tsx [hunk 3]
```diff
@@ -106,7 +108,9 @@ export default function ActiveBanner({
   id: string;
   onDismissed: () => void;
 }) {
-  if (id === PLUS_LAUNCH_ANNOUNCEMENT) {
+  const userData = useUserData();
+
+  if (id === PLUS_LAUNCH_ANNOUNCEMENT && isPlusAvailable(userData)) {
     return <PlusLaunchAnnouncementBanner onDismissed={onDismissed} />;
   }
   throw new Error(`Unrecognized banner to display (${id})`);
```
Comment (caugner) [inline client/src/banners/active-banner.tsx:116]: @schalkneethling If Plus is not available, won't this throw an Error?

Comment (caugner) [reply client/src/banners/active-banner.tsx]: Fixed in 190687808.

**Review by caugner — APPROVED** [review]

**Pull request merged**